### DISCUSSION TITLE
[OFF TOPIC] Did you ever had a host crash on Zen 4 CPU?

### DISCUSSION THREAD

**Discussion by SEsternO** [isc30/ryzen-gpu-passthrough-proxmox#51] [General]
Greetings!

There seems to be a problem going around with Zen 4 platform about the host crashing and hard resetting when using a host cpu model and nested vm capabilities. In fact, I encountered this problem myself on 7950X3D, except I didn't use nested VMs but reproduced this problem using just Windows 10 VM with LibreWolf (security oriented Firefox fork), which I assume uses some kind of virtualized sandboxing. Also, if I enable "Virtual Machine Platform" windows component, the VM starts to be more sluggish and whole host crashes even faster when using VM and LibreWolf. I also tested this config on Zen 3 Threadripper platform and it works just fine, no sluggushness, no hard crashes.

The problem seems to be related to VLS feature (_kvm_amd.vls=1_ flag), which is enabled by default in Proxmox. When you disable this feature the problem with hard crashes gets resolved (not with sluggishness though), however disabling VLS also supposed to have some performance penalty.

This bug isn't very well documented, the first clue I had was this [reddit post](https://www.reddit.com/r/Proxmox/comments/1cym3pl/nested_virtualization_crashing_ryzen_7000_series/). There's also [this bug in kernel.org bugzilla](https://bugzilla.kernel.org/show_bug.cgi?id=219009), but seeing as there's no response and probably won't be any for a while, I wanted to report this bug on Proxmox bugzilla as it seems to be less "congested" and have more activity. But after looking for more info and finding this repository I started to doubt myself, as I didn't find any mention of the problem in this repo full of Zen 4 platform users.

And so I'm asking - did any of you encountered this problem yourself? Thanks.

**Comment by mntyfrsh** [comment]
The github repo maintainer hasn't updated the README to reflect the fact that Proxmox on Zen 4 CPU's cause Windows 11 VM's to randomly crash when CPU Type is set to 'host'. This thread has more details - https://forum.proxmox.com/threads/win11-vm-opening-many-tabs-at-once-crashes-proxmox-host.140670/. The fix is to use CPU Type 'x86-64-v4', or really anything but 'host'. I know, because I'm running a UM780 XTX w/ Ryzen 7840HS (Zen 4) and that's what resolved my problems with Windows VM's randomly crashing.

As for the nested virtualization issue, can't comment on that.

**Comment by isc30** [comment]
Hi, I’m happy to accept PRs if you think the README can be improved
